### PR TITLE
chore: prepare tests to expire feature flag 'kms:defaultKeyPolicies'

### PR DESCRIPTION
**Pull request by nija-at**
This is a follow up to - 31b1b32.

Unit tests in the `aws-s3` and `aws-glue` modules assert
resources generated by KMS.

The '@aws-cdk/aws-kms:defaultKeyPolicies' feature flag will be
reset in CDKv2. Update the necessary tests so that they continue
to work when this happens.

Two course of actions are taken here - some tests needed to
be made more 'precise' in their assertions, and others had to be
configured to only run in CDKv1.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-s3/test/bucket.test.ts [hunk 1]
```diff
@@ -1175,165 +1101,63 @@ describe('bucket', () => {
...
-                    ],
-                    'Effect': 'Allow',
-                    'Resource': [
+      expect(stack).toHaveResource('AWS::IAM::Policy', {
+        'PolicyDocument': {
+          'Statement': [
+            {
+              'Action': [
+                's3:DeleteObject*',
+                's3:PutObject*',
```
Comment (njlynch) [inline packages/@aws-cdk/aws-s3/test/bucket.test.ts:1110]: The test is 'does *not* grant PutObjectAcl', but the Statement includes 's3.PutObject*'. I see that's what it was before though; is this test set up wrong?

File: packages/@aws-cdk/aws-s3/test/bucket.test.ts [hunk 2]
```diff
@@ -1652,17 +1476,18 @@ describe('bucket', () => {
 
     });
 
-    test('in different accounts, with a KMS Key', () => {
+    // testing this only in legacy mode. see https://github.com/aws/aws-cdk/issues/12921
```
Comment (njlynch) [inline packages/@aws-cdk/aws-s3/test/bucket.test.ts:1479]: https://github.com/aws/aws-cdk/pull/12925 :D

Comment (nija-at) [reply packages/@aws-cdk/aws-s3/test/bucket.test.ts:1479]: will rebase once merged.

File: packages/@aws-cdk/aws-s3/test/bucket.test.ts (hunk 1 above)
Comment (nija-at) [reply packages/@aws-cdk/aws-s3/test/bucket.test.ts:1110]: You're mis-reading the diff. The test is 'with KMS key has appropriate permissions for multipart uploads'

File: packages/@aws-cdk/aws-kinesis/test/stream.test.ts [hunk 3]
```diff
@@ -739,7 +681,7 @@ describe('Kinesis data streams', () => {
     });
   }),
 
-  test('grantReadWrite creates and attaches a policy with access to Stream and EncryptionKey', () => {
+  test('grantREADWrite creates and attaches a policy to the principal', () => {
```
Comment (njlynch) [inline packages/@aws-cdk/aws-kinesis/test/stream.test.ts:684]:
```suggestion
  test('grantReadWrite creates and attaches a policy to the principal', () => {
```

**Review by njlynch — APPROVED** [review]

**Pull request merged**